src/webgpu/api/operation/rendering/color_target_state.spec.ts

```diff
@@ -621,6 +622,104 @@ g.test('blend_constant,not_inherited')
     t.expectOK(result);
   });
 
+g.test('color_write_mask,channel_work')
+  .desc(`Test that the color write mask works alone or with multiple channels.`)
+  .params(u =>
+    u
+      .combine('mask1', [
+        GPUConst.ColorWrite.RED,
+        GPUConst.ColorWrite.GREEN,
+        GPUConst.ColorWrite.BLUE,
+        GPUConst.ColorWrite.ALPHA,
+      ])
+      .combine('mask2', [
+        GPUConst.ColorWrite.RED,
+        GPUConst.ColorWrite.GREEN,
+        GPUConst.ColorWrite.BLUE,
+        GPUConst.ColorWrite.ALPHA,
+      ])
+  )
+  .fn(async t => {
+    const { mask1, mask2 } = t.params;
+
+    const format = 'rgba8unorm';
+    const kSize = 1;
+    const masks = [mask1, mask2];
+
+    // This blend component always shows the src values.
+    const blendComponent = { srcFactor: 'one', dstFactor: 'one', operation: 'add' } as const;
+
+    let r = 0,
+      g = 0,
+      b = 0,
+      a = 0;
+    for (let i = 0; i < 2; i++) {
+      switch (masks[i]) {
+        case GPUColorWrite.RED:
+          r = 1;
+          break;
+        case GPUColorWrite.GREEN:
+          g = 1;
+          break;
+        case GPUColorWrite.BLUE:
+          b = 1;
+          break;
+        case GPUColorWrite.ALPHA:
+          a = 1;
+          break;
+      }
+    }
+
+    const testPipeline = t.createRenderPipelineForTest({
+      format,
+      blend: { color: blendComponent, alpha: blendComponent },
+      writeMask: masks[0] | masks[1],
+    });
+
+    const renderTarget = t.device.createTexture({
+      usage: GPUTextureUsage.RENDER_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+      size: [kSize, kSize],
+      format,
+    });
+
+    const kBaseColorData = new Float32Array([32, 64, 128, 192]);
+
+    const commandEncoder = t.device.createCommandEncoder();
+    {
+      const renderPass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: renderTarget.createView(),
+            loadOp: 'load',
+            storeOp: 'store',
+          },
+        ],
+      });
+      renderPass.setPipeline(testPipeline);
+      renderPass.setBindGroup(
+        0,
+        t.createBindGroupForTest(testPipeline.getBindGroupLayout(0), kBaseColorData)
+      );
+      // Draw [1,1,1,1] with `src * 1 + dst * 1`. So the
+      // result is `[1,1,1,1] * [1,1,1,1] + [0,0,0,0] * [1,1,1,1]` = [1,1,1,1].
```

Comment:
(remove this comment if you disable blending)

Reply:
Done.